bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5938,6 +6043,205 @@ private static void closeStatement(Statement statement) {
 	}
 
 	/**
+	 * Sets the end date for the given ingest job
+	 *
+	 * @param ingestJobId The ingest job to set the end date for
+	 * @param endDate     The end date
+	 *
+	 * @throws TskCoreException If inserting into the database fails
+	 * @throws TskDataException If the ingest job is not found, or already has
+	 *                          an end time.
+	 */
+	void setIngestJobEndDateTime(int ingestJobId, long endDateTime) throws TskCoreException, TskDataException {
+		CaseDbConnection connection = connections.getConnection();
```

Comment:
Please remove the SELECT to check for the record being present and the TskDataException.